api/v2_search_filters.go

```diff
@@ -54,3 +54,42 @@ type TimeFilter struct {
 	StartTime *time.Time `json:"startTime,omitempty"`
 	EndTime   *time.Time `json:"endTime,omitempty"`
 }
+
+type SearchResponse interface {
+	GetDataLength() int
+}
+
+type SearchableFilter interface {
+	GetTimeFilter() *TimeFilter
+	SetStartTime(*time.Time)
+	SetEndTime(*time.Time)
+}
+
+type search func(response interface{}, filters SearchableFilter) error
+
+// WindowedSearch performs a new search of a specific time frame size,
+// until response data is found or the max searchable days is reached
+func WindowedSearch(fn search, size int, max int, response SearchResponse, filter SearchableFilter) error {
+	for i := 0; i < max; i += size {
+		err := fn(&response, filter)
```

Comment:
Are we not overriding the response every time we run this function?

Reply:
Yes, we are overwriting the response. But this is because the response result is empty. 

Example:
We're performing a search for a value "x"
We make a search request in the last 7 days, "x" is not found the response is empty `{ "data": [] }`
So we search again, this time moving the timeframe to start -14 days and end - 7 days
A result is found in that window, and then we return it.